packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeStablecoinPayment.sol

```diff
@@ -0,0 +1,289 @@
+    function makePayment(
+        address _stablecoin,
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        uint64 _days
+    )
+        external
+        override
+        isOptedIn(_airnode)
+        isSupportedERC20(_stablecoin, _chainId, _airnode)
+    {
+        require(_chainId != 0, ERROR_ZERO_CHAINID);
+        require(_airnode != address(0), ERROR_ZERO_ADDRESS);
+        require(_requesterAddress != address(0), ERROR_ZERO_ADDRESS);
+        require(
+            _days * 1 days <= maximumWhitelistingDuration,
+            "Exceed maximum whitelisting"
+        );
+
+        (
+            uint64 expirationTimestamp,
+            uint192 indefiniteWhitelistCount
+        ) = IRequesterAuthorizerWithManager(
+                chainIdToRequesterAuthorizerWithManager[_chainId]
+            ).airnodeToEndpointIdToRequesterToWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress
+                );
+
+        require(
+            indefiniteWhitelistCount == 0,
+            "Requester already indefinently whitelisted"
+        );
+        require(
+            _days * 1 days + expirationTimestamp <= maximumWhitelistingDuration,
+            "Exceed maximum whitelisting"
+        );
+
+        uint256 paymentAmount = getPaymentAmount(
+            _stablecoin,
+            _chainId,
+            _airnode,
+            _endpointId,
+            _days
+        );
+
+        require(
+            IERC20Extended(_stablecoin).balanceOf(msg.sender) >= paymentAmount,
+            ERROR_INSUFFICIENT_AMOUNT
+        );
+
+        IRequesterAuthorizerWithManager(
+            chainIdToRequesterAuthorizerWithManager[_chainId]
+        ).setWhitelistExpiration(
+                _airnode,
+                _endpointId,
+                _requesterAddress,
+                _days * 1 days + expirationTimestamp
+            );
+
+        assert(
+            IERC20Extended(_stablecoin).transferFrom(
+                msg.sender,
+                airnodePaymentAddress[_airnode],
+                paymentAmount
+            )
+        );
+
+        emit MadePayment(
+            _chainId,
+            _airnode,
+            _endpointId,
+            _requesterAddress,
+            msg.sender,
+            airnodePaymentAddress[_airnode],
+            paymentAmount,
+            _days * 1 days + expirationTimestamp
+        );
+    }
+
+    /// @notice Returns the transfer amount a sponsor has to
+    /// transfer for a given chainId-airnode-endpointId
+    /// @param _chainId The id of the chain
+    /// @param _airnode The airnode address
+    /// @param _endpointId The endpointId
+    function getPaymentAmount(
+        address _stablecoin,
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        uint64 _days
```

Comment:
Since this is a public function shouldn't we also add checks like the following one here:

```
        require(
            _days * 1 days <= maximumWhitelistingDuration,
            "Exceed maximum whitelisting"
        );
```

Reply:
Hm maybe, Its more to get the price more than anything, so I don't think that check is really necessary